internal/fs/tracing_test.go

```diff
@@ -0,0 +1,116 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fs_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/v3/cfg"
+	"github.com/googlecloudplatform/gcsfuse/v3/internal/fs"
+	"github.com/googlecloudplatform/gcsfuse/v3/internal/fs/wrappers"
+	"github.com/googlecloudplatform/gcsfuse/v3/internal/storage/fake"
+	"github.com/googlecloudplatform/gcsfuse/v3/internal/storage/gcs"
+	"github.com/jacobsa/fuse/fuseops"
+	"github.com/jacobsa/fuse/fuseutil"
+	"github.com/jacobsa/timeutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func createTestFileSystemWithTraces(ctx context.Context, t *testing.T, ignoreInterrupts bool) (gcs.Bucket, fuseutil.FileSystem) {
+	t.Helper()
+
+	bucketName := "test-bucket"
+	bucket := fake.NewFakeBucket(timeutil.RealClock(), bucketName, gcs.BucketType{Hierarchical: false})
+	serverCfg := &fs.ServerConfig{
+		NewConfig: &cfg.Config{
+			Write: cfg.WriteConfig{
+				GlobalMaxBlocks: 1,
+			},
+			Read: cfg.ReadConfig{
+				GlobalMaxBlocks:    1,
+				BlockSizeMb:        1,
+				MaxBlocksPerHandle: 10,
+			},
+			FileSystem: cfg.FileSystemConfig{
+				IgnoreInterrupts: ignoreInterrupts,
+			},
+			Monitoring: cfg.MonitoringConfig{
+				ExperimentalTracingMode:          "stdout",
+				ExperimentalTracingSamplingRatio: 1.0,
+			},
+		},
+		CacheClock: &timeutil.SimulatedClock{},
+		BucketName: bucketName,
+		BucketManager: &fakeBucketManager{
+			buckets: map[string]gcs.Bucket{
+				bucketName: bucket,
+			},
+		},
+		SequentialReadSizeMb: 200,
+	}
+	server, err := fs.NewFileSystem(ctx, serverCfg)
+	require.NoError(t, err, "NewFileSystem")
+	return bucket, server
+}
+
+func newInMemoryExporter(t *testing.T) *tracetest.InMemoryExporter {
+	t.Helper()
+	ex := tracetest.NewInMemoryExporter()
+	otel.SetTracerProvider(sdktrace.NewTracerProvider(sdktrace.WithSyncer(ex)))
+	return ex
+}
+
+func runTestWithLookUpNode(t *testing.T, bucket gcs.Bucket, server fuseutil.FileSystem) {
+	ex := newInMemoryExporter(t)
+	t.Cleanup(func() {
+		ex.Reset()
+	})
+
+	ctx := context.Background()
+	fileName := "test.txt"
+	content := "test content"
+	createWithContents(ctx, t, bucket, fileName, content)
+
+	lookupOp := &fuseops.LookUpInodeOp{
+		Parent: fuseops.RootInodeID,
+		Name:   fileName,
+	}
+	m := wrappers.WithTracing(server)
+	err := m.LookUpInode(context.Background(), lookupOp)
+	require.NoError(t, err)
+
+	ss := ex.GetSpans()
+	require.Len(t, ss, 1)
+	assert.Equal(t, "LookUpInode", ss[0].Name)
+	assert.Equal(t, trace.SpanKindServer, ss[0].SpanKind)
+}
+
+func TestTraceLookupInode(t *testing.T) {
+	ctx := context.Background()
+	bucket, server := createTestFileSystemWithTraces(ctx, t, true)
+
+	runTestWithLookUpNode(t, bucket, server)
```

Comment:
Please inline the test and don't hide it away in a separate method. You can use [table-driven](https://go.dev/wiki/TableDrivenTests) tests for code reuse. Apart from testing, tests are useful to understand the functionality. What tests are doing should also be easily apparent and the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), which is very useful for writing business logic should take a backseat for tests.